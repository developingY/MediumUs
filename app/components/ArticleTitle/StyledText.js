import styled from 'styled-components';

const TitleWrapper = styled.div`
  max-height: 135px;
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  margin-bottom: 16px;

  overflow: hidden;
  text-overflow: ellipsis;
`;

const StyledTitle = styled.h1`
  font-size: 24px;
  font-weight: 600;
  margin-bottom: 4px;

  letter-spacing: -0.42px;
  color: rgba(0, 0, 0, 0.84);
  -webkit-line-clamp: 3;
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  @media screen and (max-width: 767px) {
    font-size: 21px;
    -webkit-line-clamp: 4;
  }
`;

const StyledSubtitle = styled.div`
  overflow: hidden;
  // max-height: 60px;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  line-height: 20px;
  color: rgba(0, 0, 0, 0.54);
`;

export { TitleWrapper, StyledTitle, StyledSubtitle };
