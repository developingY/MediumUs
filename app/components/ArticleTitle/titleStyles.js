//  wrapper : css``,
//  h1 : css``,
//  sub : css``,
import { css } from 'styled-components';

// Home page
export const HomeHeroLeft = {
  wrapper: css`
    max-height: 135px;
  `,
  h1: css`
    font-size: 24px;
    letter-spacing: -0.42px;
    margin-bottom: 4px;
    -webkit-line-clamp: 3;
    @media screen and (max-width: 767px) {
      font-size: 21px;
      -webkit-line-clamp: 4;
    }
  `,
  sub: css`
    -webkit-line-clamp: 2;
    line-height: 20px;
  `,
};

export const HomeHeroMid = {
  wrapper: css`
    max-height: 50px;
  `,
  h1: css`
    font-size: 18px;
    -webkit-line-clamp: 2;
    letter-spacing: -0.42px;
    @media screen and (max-width: 680px) {
      margin-bottom: 2px;
      -webkit-line-clamp: 2;
    }
  `,
  sub: css`
    @media screen and (max-width: 680px) {
      -webkit-line-clamp: 1;
      line-height: 20px;
    }
  `,
};

export const HomeHeroRight = {
  wrapper: css``,
  h1: css`
    font-size: 21px;
    -webkit-line-clamp: 2;
    letter-spacing: -0.42px;
    margin-bottom: 4px;
  `,

  sub: css`
    -webkit-line-clamp: 1;
    line-height: 20px;
  `,
};

export const HomeList = {
  wrapper: css`
    @media screen and (max-width: 767px) {
      max-height: 50px;
    }
  `,
  h1: css`
    font-size: 24px;
    letter-spacing: -0.42px;
    margin-bottom: 2px;
    -webkit-line-clamp: 3;
    @media screen and (max-width: 767px) {
      font-size: 18px;
      -webkit-line-clamp: 2;
    }
  `,
  sub: css`
    -webkit-line-clamp: 1;
    line-height: 20px;
  `,
};

// Homepage sidebar and TopicHomepage Sidebar are same
export const Sidebar = {
  wrapper: css``,
  h1: css`
    font-size: 18px;
    -webkit-line-clamp: 3;
    letter-spacing: -0.17px;
  `,
  sub: css`
    display: none;
  `,
};

// Topic Home Page
export const TopicHomepageHero = {
  wrapper: css``,
  h1: css`
    font-size: 44px;
    line-height: 48px;
    -webkit-line-clamp: 2;
    letter-spacing: -1.25px;
    @media screen and (max-width: 728px) {
      font-size: 26px;
      line-height: 28px;
      letter-spacing: -0.47px;
      -webkit-line-clamp: 2;
    }
  `,
  sub: css`
    font-size: 18px;
    line-height: 24px;
    -webkit-line-clamp: 2;
  `,
};

export const TopicHomepageList = {
  wrapper: css``,
  h1: css`
    font-size: 22px;
    line-height: 24px;
    letter-spacing: -0.32px;
    -webkit-line-clamp: 2;
  `,
  sub: css`
    font-size: 18px;
    line-height: 24px;
    font-weight: 300;
    -webkit-line-clamp: 2;
  `,
};

// Collection Home page
export const CollectionHomepageHero = {
  wrapper: css``,
  h1: css`
    font-size: 32px;
    -webkit-line-clamp: 4;
    line-height: 1.1;
  `,
  sub: css`
    line-height: 1.2;
    font-size: 20px;
    font-weight: 400;
    -webkit-line-clamp: 3;
  `,
};

export const CollectionHomepageList = {
  wrapper: css``,
  h1: css`
    font-size: 26px;
    -webkit-line-clamp: 3;
    line-height: 1.1;
    margin-bottom: 6px;
  `,
  sub: css`
    line-height: 1.2;
    font-size: 20px;
    font-weight: 400;
    -webkit-line-clamp: 3;
    @media screen and (max-width: 767px) {
      -webkit-line-clamp: 2;
      margin-top: 1px;
    }
  `,
};

// Article Page
export const ArticlePageTitle = {
  wrapper: css``,
  h1: css`
    line-height: 48px;
    font-size: 40px;
    font-family: medium-content-title-font, Georgia, Cambria, 'Times New Roman',
      Times, serif;
    font-weight: 400;
    margin-bottom: 8px;
  `,
  sub: css`
    line-height: 32px;
    font-size: 24px;
    font-weight: 300;
  `,
};

export const ArticlePageRecommend = {
  wrapper: css``,
  h1: css`
    line-height: 32px;
    font-size: 24px;
    font-family: medium-content-title-font, Georgia, Cambria, 'Times New Roman',
      Times, serif;
    font-weight: 500;
    -webkit-line-clamp: 4;
  `,
  sub: css`
    display: none;
  `,
};
