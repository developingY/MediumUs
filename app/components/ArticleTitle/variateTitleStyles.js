import { css } from 'styled-components';

// Home page
export const HomeHeroLeft = {
  wrapper: css`
    max-height: 135px;
    margin-top: 16px;
    margin-bottom: 16px;
  `,
  h1: css`
    font-size: 24px;
    letter-spacing: -0.42px;
    margin-bottom: 4px;
    -webkit-line-clamp: 3;
    @media screen and (max-width: 767px) {
      font-size: 21px;
      -webkit-line-clamp: 3;
    }
  `,
  sub: css`
    -webkit-line-clamp: 3;
    line-height: 20px;
  `,
};

export const HomeHeroMid = {
  wrapper: css`
    max-height: 50px;
    margin-bottom: 16px;
  `,
  h1: css`
    font-size: 18px;
    -webkit-line-clamp: 2;
    letter-spacing: -0.42px;
    margin-bottom: 4px;
    @media screen and (max-width: 680px) {
      margin-bottom: 2px;
      -webkit-line-clamp: 2;
    }
  `,
  sub: css`
    -webkit-line-clamp: 1;
    @media screen and (max-width: 680px) {
      line-height: 20px;
    }
  `,
};

export const HomeHeroRight = {
  wrapper: css`
    margin-top: 16px;
    margin-bottom: 16px;
  `,
  h1: css`
    font-size: 21px;
    -webkit-line-clamp: 2;
    letter-spacing: -0.42px;
    margin-bottom: 4px;
  `,

  sub: css`
    -webkit-line-clamp: 1;
    line-height: 20px;
  `,
};

export const HomeList = {
  wrapper: css`
    @media screen and (max-width: 767px) {
      max-height: 62px;
    }
    margin-bottom: 12px;
  `,
  h1: css`
    font-size: 24px;
    letter-spacing: -0.42px;
    margin-bottom: 2px;
    -webkit-line-clamp: 3;
    @media screen and (max-width: 767px) {
      font-size: 18px;
      -webkit-line-clamp: 2;
    }
  `,
  sub: css`
    -webkit-line-clamp: 1;
    line-height: 20px;
  `,
};

// Topic Home Page
export const TopicHomepageHero = {
  wrapper: css`
    margin-top: 13px;
    @media screen and (max-width: 728px) {
      margin-top: 21px;
    }
    margin-bottom: 16px;
  `,
  h1: css`
    font-size: 44px;
    line-height: 48px;
    -webkit-line-clamp: 2;
    letter-spacing: -1.25px;
    margin-bottom: 4px;
    @media screen and (max-width: 728px) {
      font-size: 25.2px;
      line-height: 28px;
      letter-spacing: -0.47px;
      -webkit-line-clamp: 2;
    }
  `,
  sub: css`
    font-size: 18px;
    line-height: 24px;
    -webkit-line-clamp: 2;
  `,
};

export const TopicHomepageList = {
  wrapper: css`
    margin-bottom: 12px;
  `,
  h1: css`
    font-size: 21.6px;
    line-height: 24px;
    letter-spacing: -0.32px;
    -webkit-line-clamp: 2;
    margin-bottom: 4px;
    @media screen and (max-width: 855px) {
      margin-bottom: 0px;
    }
  `,
  sub: css`
    font-size: 18px;
    line-height: 24px;
    font-weight: 300;
    -webkit-line-clamp: 2;
  `,
};

// Publication Home page
export const PublicationHomepageHero = {
  wrapper: css`
    margin-top: 7px;
    margin-bottom: 20px;
  `,
  h1: css`
    font-size: 32px;
    -webkit-line-clamp: 4;
    line-height: 1.1;
    @media screen and (max-width: 767px) {
      font-size: 26px;
      margin-top: 15px;
      -webkit-line-clamp: 3;
    }
  `,
  sub: css`
    line-height: 1.2;
    font-size: 20px;
    font-weight: 400;
    -webkit-line-clamp: 3;
  `,
};

export const PublicationHomepageList = {
  wrapper: css`
    margin-top: 20px;
    margin-bottom: 20px;
  `,
  h1: css`
    font-size: 26px;
    -webkit-line-clamp: 3;
    line-height: 1.1;
    margin-bottom: 6px;
  `,
  sub: css`
    line-height: 1.2;
    font-size: 20px;
    font-weight: 400;
    -webkit-line-clamp: 3;
    @media screen and (max-width: 767px) {
      -webkit-line-clamp: 2;
      margin-top: 1px;
    }
  `,
};
export const PublicationHomepageListX2 = PublicationHomepageList;
export const PublicationHomepageListX3 = PublicationHomepageList;

// Article Page
export const ArticlePageTitle = {
  wrapper: css`
    margin-bottom: 20px;
  `,
  h1: css`
    line-height: 48px;
    font-size: 40px;
    font-family: vanilla-fell, medium-content-title-font, Georgia, Cambria,
      'Times New Roman', Times, serif;
    font-weight: 400;
    margin-top: 0.78em;
    margin-bottom: 8px;
    color: rgba(0, 0, 0, 0.84);
    @media screen and (max-width: 728px) {
      font-size: 30px;
      line-height: 40px;
    }
  `,
  sub: css`
    line-height: 32px;
    font-size: 24px;
    font-weight: 300;
    color: rgba(0, 0, 0, 0.54);
    font-family: vanilla-sohne, medium-content-title-font, serif;
    @media screen and (max-width: 728px) {
      font-size: 16px;
      line-height: 22px;
    }
  `,
};

export const ArticlePageRecommendation = {
  wrapper: css`
    margin-bottom: 16px;
  `,
  h1: css`
    line-height: 32px;
    font-size: 24px;
    font-family: medium-content-title-font, Georgia, Cambria, 'Times New Roman',
      Times, serif;
    font-weight: 500;
    -webkit-line-clamp: 4;
  `,
  sub: css`
    display: none;
  `,
};

// Homepage sidebar and TopicHomepage Sidebar are same
export const Sidebar = {
  wrapper: css``,
  h1: css`
    font-size: 18px;
    -webkit-line-clamp: 3;
    letter-spacing: -0.17px;
  `,
  sub: css`
    display: none;
  `,
};
