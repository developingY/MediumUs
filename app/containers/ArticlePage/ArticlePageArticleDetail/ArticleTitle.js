import React from 'react';
import PropTypes from 'prop-types';
import { TitleWrapper } from './Wrappers';
import ArticlePoster from '../../../components/ArticlePoster';

function ArticleTitle(props) {
  const id = props.children[0];

  return (
    <TitleWrapper>
      <ArticlePoster id={id} variation="ArticlePageTitle" />
    </TitleWrapper>
  );
}

ArticleTitle.propTypes = {
  children: PropTypes.array.isRequired,
};

export default ArticleTitle;
