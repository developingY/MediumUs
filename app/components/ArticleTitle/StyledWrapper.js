import styled from 'styled-components';
import * as variateTitleStyles from './variateTitleStyles';

const Wrapper = styled.div`
  display: flex;
  flex-flow: column wrap;
  word-wrap: break-word;
  overflow: hidden;
  ${props => variateTitleStyles[props.variation].wrapper}
`;

const StyledTitle = styled.h1`
  font-weight: 600;
  color: rgba(0, 0, 0, 0.84);
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  margin: 0;
  ${props => variateTitleStyles[props.variation].h1}
`;

const StyledSubtitle = styled.div`
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  color: rgba(0, 0, 0, 0.54);
  ${props => variateTitleStyles[props.variation].sub}
`;

export const PublicationColumn = styled.h2`
  margin-top: 30px;
  margin-bottom: -20px;
  letter-spacing: 0.05em;
  text-transform: uppercase;
  font-size: 15px;
  font-weight: 300;
  color: rgba(0, 0, 0, 0.54);
`;

export { Wrapper, StyledTitle, StyledSubtitle };
