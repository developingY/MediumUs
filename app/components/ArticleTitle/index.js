/**
 *
 * ArticleTitle
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { TitleWrapper, StyledTitle, StyledSubtitle } from './StyledWrapper';
import * as titleStyles from './titleStyles';

function ArticleTitle(props) {
  return (
    <TitleWrapper {...props}>
      <a href={props.articleLink}>
        <StyledTitle {...props}>{props.title} </StyledTitle>
      </a>
      <a href={props.articleLink}>
        <StyledSubtitle {...props}>{props.subtitle}</StyledSubtitle>
      </a>
    </TitleWrapper>
  );
}

ArticleTitle.propTypes = {
  articleLink: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  type: PropTypes.oneOf([..._.keys(titleStyles)]).isRequired,
  // type: PropTypes.string.isRequired,
};

export default ArticleTitle;
