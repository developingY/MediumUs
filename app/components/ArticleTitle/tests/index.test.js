/**
 *
 * Tests for ArticleTitle
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import 'jest-dom/extend-expect';
import articleInfo from '../stories/exampleData';
import ArticleTitle from '../index';

describe('<ArticleTitle />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<ArticleTitle {...articleInfo} variation="HomeHeroLeft" />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<ArticleTitle {...articleInfo} variation="ArticlePageTitle" />);
    expect(firstChild).toMatchSnapshot();
  });

  it('Should have title and subtitle content', () => {
    const { container } = render(
      <ArticleTitle {...articleInfo} variation="TopicHomepageHero" />,
    );
    expect(container).toHaveTextContent(articleInfo.title);
    expect(container).toHaveTextContent(articleInfo.subtitle);
  });

  it('Should have no subtitle content', () => {
    const { container } = render(
      <ArticleTitle {...articleInfo} variation="Sidebar" />,
    );
    expect(container.querySelectorAll('div')[1]).toHaveTextContent('');
  });
});
