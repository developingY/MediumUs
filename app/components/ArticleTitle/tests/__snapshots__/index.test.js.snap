// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleTitle /> Should render and match the snapshot 1`] = `
<div
  class="StyledWrapper__TitleWrapper-sc-111jh54-0 ealqyf"
  title="Face Filters for Instagram and Snapchat Are the New Frontier of Surrealist Art"
  type="ArticlePageTitle"
>
  <a
    href="./"
  >
    <h1
      class="StyledWrapper__StyledTitle-sc-111jh54-1 gYdKdB"
      title="Face Filters for Instagram and Snapchat Are the New Frontier of Surrealist Art"
      type="ArticlePageTitle"
    >
      Face Filters for Instagram and Snapchat Are the New Frontier of Surrealist Art
       
    </h1>
  </a>
  <a
    href="./"
  >
    <div
      class="StyledWrapper__StyledSubtitle-sc-111jh54-2 bomnVF"
      title="Face Filters for Instagram and Snapchat Are the New Frontier of Surrealist Art"
      type="ArticlePageTitle"
    >
      And one last warning about their stupidly popular little brother
    </div>
  </a>
</div>
`;
